test/cljam/t_bam.clj

```diff
@@ -40,23 +45,96 @@
+  (when (= (count alns1) (count alns2))
+    (loop [alns1 alns1
+           alns2 alns2]
+      (let [aln1 (first alns1)
+            aln2 (first alns2)]
+        (if (nil? aln1)
+          true
+          (when (and (= (:rname aln1) (:rname aln2))
+                     (= (:pos aln1) (:pos aln2)))
+            (recur (rest alns1) (rest alns2))))))))
+
+(defn- pointer= [alns1 alns2]
+  (when (= (count alns1) (count alns2))
+    (loop [alns1 alns1
+           alns2 alns2]
+      (let [aln1 (first alns1)
+            aln2 (first alns2)]
+        (if (nil? aln1)
+          true
+          (when (and (= (:rname aln1) (:rname aln2))
+                     (= (:pos aln1) (:pos aln2))
+                     (= (:flag aln1) (:flag aln2)))
+            (recur (rest alns1) (rest alns2))))))))
+
 (deftest bamreader
   (with-before-after {:before (do (prepare-cache!)
                                   (spit-bam-for-test temp-file test-sam))
                       :after (clean-cache!)}
-    (let [rdr (bam/reader temp-file :ignore-index true)]
-      (is (= (io/read-refs rdr) test-sam-refs)))))
+    (let [rdr (bam/reader temp-file :ignore-index false)]
+      (is (= (io/read-refs rdr) test-sam-refs))
+      (is (= (io/read-alignments rdr) (:alignments test-sam))))
+    (let [rdr (bam/reader temp-file :ignore-index false)]
+      (is (= (io/read-refs rdr) test-sam-refs))
+      (is (thrown? Exception (io/read-alignments rdr {:chr "ref2"}))))
+    (let [rdr (bam/reader temp-file :ignore-index false)]
+      (is (= (io/read-refs rdr) test-sam-refs))
+      (is (= (io/read-alignments rdr {:depth :deep}) (:alignments test-sam))))
+    (let [rdr (bam/reader temp-file :ignore-index false)]
+      (is (= (io/read-refs rdr) test-sam-refs))
+      (is (shallow= (io/read-alignments rdr {:depth :shallow})
+                    (:alignments test-sam))))
+    (let [rdr (bam/reader temp-file :ignore-index false)]
+      (is (= (io/read-refs rdr) test-sam-refs))
+      (is (pointer= (io/read-alignments rdr {:depth :pointer})
+                    (:alignments test-sam))))
+    (let [rdr (bam/reader temp-file :ignore-index false)]
+      (is (= (io/read-refs rdr) test-sam-refs))
+      (is (= (data->clj (io/read-blocks rdr)) test-sam-data)))
+    (let [rdr (bam/reader temp-file :ignore-index false)]
+      (is (= (io/read-refs rdr) test-sam-refs))
+      (is (thrown? Exception (data->clj (io/read-blocks rdr {:chr "ref2"})))))))
 
-(deftest-slow bamreader-medium-file
-  (with-before-after {:before (do (prepare-cache!)
-                                  (spit-bam-for-test temp-file test-sam))
-                      :after (clean-cache!)})
-    (let [rdr (bam/reader medium-bam-file :ignore-index true)]
-      (is (= (io/read-refs rdr) medium-sam-refs))))
+(deftest bamreader-with-index
+  (with-before-after {:before (prepare-cache!)
+                      :after (clean-cache!)}
+    (let [rdr (bam/reader test-sorted-bam-file :ignore-index false)]
+      ;; TODO: Does not works?
```

Comment:
All right, I will open issue for this, too.
I guess I wrong something, but I didn't know this reason.